options.go

```diff
@@ -54,6 +54,13 @@ func Hooks(hooks ...func(zapcore.Entry) error) Option {
 	})
 }
 
+// HooksWithFields similiar with Hooks, but hook funcs can access fields
```

Comment:
```suggestion
// HooksWithFields is similar to Hooks, but allows access to fields in addition to the entry.
```

we should also mention include documentation that the fields are read-only (fields can't be changed or added), as well as note that there's a performance impact as it requires all fields to be stored separately (in addition to the encoded form)